### PR TITLE
cloudflared: 2022.4.0 -> 2022.4.1 + fix tests for linux

### PR DESCRIPTION
**Pull request by superherointj**
cloudflared: 2022.4.0 -> 2022.4.1

* fix tests for linux

### CODE REVIEW

File: pkgs/applications/networking/cloudflared/default.nix [hunk 1]
```diff
@@ -13,7 +13,13 @@ buildGoModule rec {
 
   vendorSha256 = null;
 
-  doCheck = false;
+  postPatch = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cloudflared/default.nix:16]: Please add some comments why the tests fail.

Comment (superherointj) [reply pkgs/applications/networking/cloudflared/default.nix]: Done

File: pkgs/applications/networking/cloudflared/default.nix [hunk 2]
```diff
@@ -13,10 +13,21 @@ buildGoModule rec {
...
 
-  doCheck = false;
-
   ldflags = [ "-X main.Version=${version}" ];
 
+  preCheck = ''
+    # Workaround for: sshgen_test.go:74:
+    #   mkdir /homeless-shelter/.cloudflared: no such file or directory
+
+    export HOME=".";
```
[the remark this answers is not in the document]
Comment (superherointj) [reply pkgs/applications/networking/cloudflared/default.nix]: Done

File: pkgs/applications/networking/cloudflared/default.nix [hunk 3]
```diff
@@ -13,10 +13,23 @@ buildGoModule rec {
...
+    export HOME="''$(mktemp -d)";
+
+    # Workaround for: protocol_test.go:11:
+    #   lookup protocol-v2.argotunnel.com on [::1]:53: read udp [::1]:51876->[::1]:53: read: connection refused
+
+    substituteInPlace "edgediscovery/protocol_test.go" \
+      --replace "TestProtocolPercentage" "SkipProtocolPercentage"
+  '';
+
+  doCheck = !stdenv.isDarwin;
```
Comment (06kellyjac) [inline pkgs/applications/networking/cloudflared/default.nix:31]: import `stdenv` at the top

Comment (superherointj) [reply pkgs/applications/networking/cloudflared/default.nix:31]: Done

**Comment by techknowlogick** [comment]
The code changes lgtm, but I'm on darwin so I can test them. Looks like ofborg may be failing on `x86_64-linux` though https://logs.nix.ci/?key=nixos/nixpkgs.167419&attempt_id=4423dd5c-2407-4d57-bc1f-dab63e93ff05 I think it is a networking related test and so could probably be skipped.

I'll leave it to the others to give a formal approval of this PR.

**Comment by superherointj** [comment]
> The code changes lgtm, but I'm on darwin so I can test them.

Feel free to propose a fix for darwin.

**Comment by techknowlogick** [comment]
> Feel free to propose a fix for darwin.

I will, but probably won't be able to make it into this PR

**Comment by superherointj** [comment]
> Looks like ofborg may be failing on `x86_64-linux` though https://logs.nix.ci/?key=nixos/nixpkgs.167419&attempt_id=4423dd5c-2407-4d57-bc1f-dab63e93ff05 I think it is a networking related test and so could probably be skipped.

On my computer, it pass:
image

**Comment by superherointj** [comment]
https://github.com/ofborg eval

**Comment by superherointj** [comment]
@ofborg eval

**Comment by superherointj** [comment]
@ofborg eval

**Comment by 06kellyjac** [comment]
I'm not sure why the rest run fine on ofborg but tcp-ws_proxy case in particular fails

https://github.com/cloudflare/cloudflared/blob/master/proxy/proxy_test.go#L481-L643

I guess first thing would be to rebase this on 2022.4.1 and see if the issue persists.

Maybe it's a threading thing and the destination address is busy? :shrug:

**Comment by superherointj** [comment]
> I'm not sure why the rest run fine on ofborg but tcp-ws_proxy case in particular fails
> 
> https://github.com/cloudflare/cloudflared/blob/master/proxy/proxy_test.go#L481-L643
> 
> I guess first thing would be to rebase this on 2022.4.1 and see if the issue persists.
> 
> Maybe it's a threading thing and the destination address is busy? shrug

Rebased to 2022.4.1. Tests aren't failing for me on my host.

**Comment by superherointj** [comment]
> I'm not sure why the rest run fine on ofborg but tcp-ws_proxy case in particular fails

Ofborg did not fail on second run.

**Comment by 06kellyjac** [comment]
> Ofborg pass on 2nd run.

Was that before the update?

We should keep an eye on it but if it's a flakey test maybe we should just skip it.

File: pkgs/applications/networking/cloudflared/default.nix [hunk 4]
```diff
@@ -1,22 +1,35 @@
...
 
-  doCheck = false;
-
   ldflags = [ "-X main.Version=${version}" ];
 
+  preCheck = ''
+    # Workaround for: sshgen_test.go:74:
+    #   mkdir /homeless-shelter/.cloudflared: no such file or directory
+
+    export HOME="''$(mktemp -d)";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cloudflared/default.nix:22]:
```suggestion
    # Workaround for: sshgen_test.go:74: mkdir /homeless-shelter/.cloudflared: no such file or directory
    export HOME="$(mktemp -d)";
```
thats pretty common.

**Comment by superherointj** [comment]
> > Ofborg pass on 2nd run.
> 
> Was that before the update?

Yes.

> We should keep an eye on it but if it's a flakey test maybe we should just skip it.

Too early to tell. It failed once. I could not reproduce it.

**Pull request merged**